src/rotator_library/anthropic_compat/streaming_fast.py

```
@@ -0,0 +1,460 @@
+            try:
+                chunk = json_loads(data_content)
+            except Exception:
+                continue
+
+            # Extract usage if present
+            if "usage" in chunk and chunk["usage"]:
+                usage = chunk["usage"]
+                # Provider returned usage - use it (overrides precomputed)
+                if usage.get("prompt_tokens"):
+                    input_tokens = usage.get("prompt_tokens", input_tokens)
+                    usage_received_from_provider = True
```

Ruff (0.15.4)

[error] Local variable `usage_received_from_provider` is assigned to but never used

Remove assignment to unused variable `usage_received_from_provider`

(F841)